debezium-server/debezium-server-rocketmq/src/main/java/io/debezium/server/rocketmq/RocketMqChangeConsumer.java

```diff
@@ -0,0 +1,191 @@
+            mqProducer = customRocketMqProducer.get();
+            startProducer();
+            LOGGER.info("Obtained custom configured RocketMqProducer '{}'", mqProducer);
+            return;
+        }
+
+        final Config config = ConfigProvider.getConfig();
+        // init rocketmq producer
+        RPCHook rpcHook = null;
+        Optional<Boolean> aclEnable = config.getOptionalValue(PROP_PRODUCER_ACL_ENABLE, Boolean.class);
+        if (aclEnable.isPresent() && aclEnable.get()) {
+            if (config.getOptionalValue(PROP_PRODUCER_ACCESS_KEY, String.class).isEmpty()
+                    || config.getOptionalValue(PROP_PRODUCER_SECRET_KEY, String.class).isEmpty()) {
+                throw new DebeziumException("When acl.enabled is true, access key and secret key cannot be empty");
+            }
+            rpcHook = new AclClientRPCHook(
+                    new SessionCredentials(
+                            config.getValue(PROP_PRODUCER_ACCESS_KEY, String.class),
+                            config.getValue(PROP_PRODUCER_SECRET_KEY, String.class)));
+        }
+        this.mqProducer = new DefaultMQProducer(rpcHook);
+        this.mqProducer.setNamesrvAddr(config.getValue(PROP_PRODUCER_NAME_SRV_ADDR, String.class));
+        this.mqProducer.setInstanceName(createUniqInstance(config.getValue(PROP_PRODUCER_NAME_SRV_ADDR, String.class)));
+        this.mqProducer.setProducerGroup(config.getValue(PROP_PRODUCER_GROUP, String.class));
+
+        if (config.getOptionalValue(PROP_PRODUCER_SEND_MSG_TIMEOUT, Integer.class).isPresent()) {
+            this.mqProducer.setSendMsgTimeout(config.getValue(PROP_PRODUCER_SEND_MSG_TIMEOUT, Integer.class));
+        }
+
+        if (config.getOptionalValue(PROP_PRODUCER_MAX_MESSAGE_SIZE, Integer.class).isPresent()) {
+            this.mqProducer.setMaxMessageSize(config.getValue(PROP_PRODUCER_MAX_MESSAGE_SIZE, Integer.class));
+        }
+
+        this.mqProducer.setLanguage(LanguageCode.JAVA);
+        startProducer();
+    }
+
+    private void startProducer() {
+        try {
+            this.mqProducer.start();
+            LOGGER.info("Consumer started...");
+        } catch (MQClientException e) {
+            throw new DebeziumException(e);
+        }
+    }
+
+    private String createUniqInstance(String prefix) {
+        return prefix.concat("-").concat(UUID.randomUUID().toString());
+    }
+
+    @PreDestroy
+    void close() {
+        // Closed rocketmq producer
+        LOGGER.info("Consumer destroy...");
+        if (mqProducer != null) {
+            mqProducer.shutdown();
+        }
+    }
+
+    @Override
+    public void handleBatch(List<ChangeEvent<Object, Object>> records, DebeziumEngine.RecordCommitter<ChangeEvent<Object, Object>> committer)
+            throws InterruptedException {
+        final CountDownLatch latch = new CountDownLatch(records.size());
+        Deque<SubmittedRecord> submittedRecords = new LinkedList<>();
```

Comment:
Is this needed? You already have a list of records in `records`.

Reply:
SubmittedRecord is to ensure the order

Reply:
you have already this order in `records` list, no need to repeat it